src/main.rs

```diff
@@ -0,0 +1,56 @@
+fn main() {
+    loop {
+        let mut terminal1 = Terminal::new();
+        let todo = terminal1.ask_for_new_todo();
+        terminal1.show_todo(&todo);
+    }
+}
+
+#[derive(Debug, Clone)]
+struct Todo {
+    message: String,
+}
+
+impl Todo {
+    fn new(message: String) -> Self {
+        Todo { message }
+    }
+}
+
+use std::io::{ Stdin, Stdout, Write };
+
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
+}
+
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
+
+    fn ask_for_new_todo(&mut self) -> Todo {
+
+        println!("🖋  Você deseja adicionar um novo TODO? (Responta do s ou n)");
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+
+        if buf.trim().to_string().to_lowercase() == "n" {
```

Comment:
```suggestion
        if buf.trim().to_lowercase() == "n" {
```

O `to_string` não é necessário aqui.

Aliás, vamos aproveitar essa mudança e já introduzir uma outra feature aqui? Faça com que o seu programa aceite _apenas_ as respostas `"s"` e `"n"`. Se receber qualquer uma sem ser elas, faça com que o usuário repita após receber um aviso como "não entendi".

Talvez você perceba valor em isolar essa funcionalidade em um outro método (por exemplo `should_ask_for_new_todo`) que retorna um booleano. Aí você deixa o `ask_for_new_todo` "limpo".